src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DecimalConverter.cs

```diff
@@ -43,34 +49,20 @@ public override bool CanConvertTo(ITypeDescriptorContext context, Type destinati
         /// </summary>
         public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
         {
-            if (destinationType == null)
+            if (destinationType == typeof(InstanceDescriptor) && value is decimal decimalValue)
             {
-                throw new ArgumentNullException(nameof(destinationType));
+                ConstructorInfo ctor = typeof(decimal).GetConstructor(new Type[] { typeof(int[]) });
+                Debug.Assert(ctor != null, "Expected constructor to exist.");
+                return new InstanceDescriptor(ctor, new object[] { decimal.GetBits(decimalValue) });
             }
 
-            if (destinationType == typeof(InstanceDescriptor) && value is decimal)
-            {
-
-                object[] args = new object[] { decimal.GetBits((decimal)value) };
-                MemberInfo member = typeof(decimal).GetConstructor(new Type[] { typeof(int[]) });
-
-                Debug.Assert(member != null, "Could not convert decimal to member. Did someone change method name / signature and not update DecimalConverter?");
-                if (member != null)
-                {
-                    return new InstanceDescriptor(member, args);
-                }
-                else
-                {
-                    return null;
-                }
-            }
-            
             return base.ConvertTo(context, culture, value, destinationType);
         }
 
         /// <summary>
         /// Convert the given value to a string using the given radix
         /// </summary>
+        [ExcludeFromCodeCoverage] // Not called as AllowHex is false.
```

Comment:
Please remove this.  If we have code that's not exercised, it should show up as such in a code coverage report.

Reply:
But it genuinely can’t be called. Decimal/double/single converter overrides `AllowHex` to return false. Therefore the convert to string overload will provably never be called because we never do hex parsing as a result. So this exclusion attribute can be justified.

Reply:
> But it genuinely can’t be called.

I understand.  But our goal isn't to get to 100% code coverage just so we can have a 100% number.  The only reason we look at code coverage is as an indicator to help drive better testing; hiding code from the report like this only serves to artificially increase the number and prevents us from seeing that we have code that's never getting executed.  If it's actually not reachable, than the fix (not for this PR) is to delete the dead code, not hide it from a code coverage report so that no one in the future will be able to see that it's not being executed.

Reply:
Very well, removed